jujugui/static/gui/src/app/app.js

```diff
@@ -929,6 +917,9 @@ YUI.add('juju-gui', function(Y) {
         return window.localStorage.getItem('discharge-token');
       };
       const charmstore = this.get('charmstore');
+      const setCreds = function(creds) {
```

Comment:
I'm surprised this works because the `this` should be pointing to the component and not `app`. I'd like some investigation done here, but I feel that you'll probably want to change this to be.

```JavaScript
const setCreds = creds => {
  this.user.controller = creds;
}
```